File: python/interview_17_16.py
class Solution:
    def massage(self, nums: List[int]) -> int:
        dp_i_0, dp_i_1 = 0, 0
        for n in nums:
            dp_i_1, dp_i_0 = n + dp_i_0, max(dp_i_0, dp_i_1)
        return max(dp_i_0, dp_i_1)

File: Java/interview_17_16.java
class Solution {
   
    public int massage(int[] nums) {
        int dp_i_0 = 0, dp_i_1 = 0;
        for (int n : nums) {
            int tmp = dp_i_0;
            dp_i_0 = Math.max(dp_i_0, dp_i_1);
            dp_i_1 = n + tmp;
        }
        return Math.max(dp_i_0, dp_i_1);
    }
}
